libraries/AP_OpenDroneID/AP_OpenDroneID.h

```diff
@@ -0,0 +1,360 @@
+    }
+
+    void set_operator_id(char* id) {
+        memset(_operator_id, 0, ODID_ID_SIZE + 1); //just in case, according to strncpy definition, the source is padded with zeroes if source length is shorter
+        strncpy(_operator_id, id, ODID_ID_SIZE + 1);
+    }
+
+    void set_description_type(MAV_ODID_DESC_TYPE type) {
+        //for now _description_type can only be MAV_ODID_DESC_TYPE_TEXT
+        _description_type = (MAV_ODID_DESC_TYPE) constrain_uint16(type, MAV_ODID_DESC_TYPE_TEXT, MAV_ODID_DESC_TYPE_TEXT);
+    }
+    void set_description(char* description) {
+        memset(_description, 0, ODID_STR_SIZE + 1); //just in case, according to strncpy definition, the source is padded with zeroes if source length is shorter
+        strncpy(_description, description, ODID_STR_SIZE + 1);
+    }
+
+    void set_operator_latitude(int32_t latitude) {
+        _operator_position.lat = latitude;
+    }
+
+    void set_operator_longitude(int32_t longitude) {
+        _operator_position.lng = longitude;
+    }
+
+    void set_operator_timestamp(time_t epoch_timestamp) {
+        uint32_t timestamp_2019 = (uint32_t) epoch_timestamp - (uint32_t) 1546300800; //1546300800 = 1/1/2019 00:00:00 UTC
+        _operator_timestamp = constrain_uint32(timestamp_2019, 0, UINT32_MAX);
+    }
+
+    void set_area_count(uint16_t count) {
+        _area_count = constrain_uint16(count, ODID_AREA_COUNT_MIN, ODID_AREA_COUNT_MAX);
+    }
+
+    void set_area_radius(uint16_t radius) {
+        _area_radius = constrain_uint16(radius, 0, ODID_MAX_AREA_RADIUS);
+    }
+
+    void set_area_ceiling(float ceiling) {
+        _area_ceiling = constrain_float(ceiling, ODID_MIN_ALT, ODID_MAX_ALT);
+    }
+
+    void set_area_floor(float floor) {
+         _area_floor = constrain_float(floor, ODID_MIN_ALT, ODID_MAX_ALT);
+    }
+
+    void set_operator_altitude_geo(float altitude){
+        altitude = constrain_float(altitude, ODID_MIN_ALT, ODID_MAX_ALT);
+        _operator_position.alt = (int32_t) altitude*100.0f;
+    }
+
+    void set_mavlink_dynamic_messages_period_ms(uint32_t period_ms){
+       _mavlink_dynamic_period_ms = constrain_int32(period_ms, 100, 1000); //between 100ms and 1 seconds, Remote ID standard specifies location data to be broadcasted every 1 second or faster.
+    }
+
+    void set_mavlink_static_messages_period_ms(uint32_t period_ms){
+       _mavlink_static_period_ms = constrain_int32(period_ms, 1000, 360000); //between 1 and 1 hour, Remote ID transponder needs to broadcast the static packages every 3 seconds or faster.
+    }
+
+    void set_mavlink_channel(mavlink_channel_t chan){
+        _chan = chan;
+    }
+
+    //transmit functions to manually send a static MAVLink message
+    void send_basic_id_message();
+    void send_system_message();
+    void send_self_id_message();
+    void send_operator_id_message();
```

Comment:
These should be `const`

```suggestion
    void send_basic_id_message() const;
    void send_system_message() const;
    void send_self_id_message() const;
    void send_operator_id_message() const;
```